File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] arr) {
       int[] ans = new int[arr.length];
       int start = 0, end = 1;

       for(int i = 0; i < arr.length; i++){
        //    if(arr.length < 2) break;

           if(arr[i] == 0){ 
            ans[ans.length - end] = 0;
            end++;
            }
            else{
                ans[start] = arr[i];
                start++;
            }
       }
       for(int i = 0; i < arr.length; i++){
           arr[i] = ans[i];
       }
}
}

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& arr) {
        int n = arr.size(), i = 0, idx = 0,j = n - 1;
        while(i < n){
            if(arr[i] == 0) {i++ ; continue;}
            arr[idx++] = arr[i++];
        }
        while(idx <= j) arr[j--] = 0;
    }
};